systemtest/src/test/java/io/strimzi/systemtest/mirrormaker/MirrorMaker2ST.java

```diff
@@ -653,6 +655,39 @@ void testMirrorMaker2CorrectlyMirrorsHeaders() {
         assertThat(log, containsString(header1));
         assertThat(log, containsString(header2));
     }
+
+    @Test
+    void testScaleMirrorMaker2ToZero() {
+        // Deploy source kafka
+        KafkaResource.kafkaEphemeral(kafkaClusterSourceName, 1, 1).done();
+        // Deploy target kafka
+        KafkaResource.kafkaEphemeral(kafkaClusterTargetName, 1, 1).done();
+
+        KafkaMirrorMaker2Resource.kafkaMirrorMaker2(CLUSTER_NAME, kafkaClusterTargetName, kafkaClusterSourceName, 3, false)
+            .editMetadata()
+                .addToLabels("type", "kafka-mirror-maker-2")
+            .endMetadata()
+            .done();
+
+        long oldObsGen = KafkaMirrorMaker2Resource.kafkaMirrorMaker2Client().inNamespace(NAMESPACE).withName(CLUSTER_NAME).get().getStatus().getObservedGeneration();
+        String mm2DepName = KafkaMirrorMaker2Resources.deploymentName(CLUSTER_NAME);
+        List<String> mm2Pods = kubeClient().listPodNames("type", "kafka-mirror-maker-2");
+        assertThat(mm2Pods.size(), is(3));
+
+        LOGGER.info("Scaling MirrorMaker to zero");
+        KafkaMirrorMaker2Resource.replaceKafkaMirrorMaker2Resource(CLUSTER_NAME, mm2 -> mm2.getSpec().setReplicas(0));
+
+        PodUtils.waitForPodsReady(kubeClient().getDeploymentSelectors(mm2DepName), 0, true);
+
+        mm2Pods = kubeClient().listPodNames("type", "kafka-mirror-maker-2");
+        KafkaMirrorMaker2Status mm2Status = KafkaMirrorMaker2Resource.kafkaMirrorMaker2Client().inNamespace(NAMESPACE).withName(CLUSTER_NAME).get().getStatus();
+        long actualObsGen = KafkaMirrorMaker2Resource.kafkaMirrorMaker2Client().inNamespace(NAMESPACE).withName(CLUSTER_NAME).get().getStatus().getObservedGeneration();
+
+        assertThat(mm2Pods.size(), is(0));
+        assertThat(mm2Status.getConditions().get(0).getType(), is("Ready"));
+        assertThat(actualObsGen, is(not(oldObsGen)));
```

Comment:
Since you already do asserts on the status, maybe you can also check that `url` is not set.